mm2src/coins/utxo/utxo_balance_events.rs

```diff
@@ -58,45 +58,29 @@ impl EventStreamer for UtxoBalanceEventStreamer {
         const RECEIVER_DROPPED_MSG: &str = "Receiver is dropped, which should never happen.";
         let streamer_id = self.streamer_id();
         let coin = self.coin;
+        let mut scripthash_to_address_map = BTreeMap::default();
 
-        async fn subscribe_to_addresses(
-            utxo: &UtxoCoinFields,
-            addresses: HashSet<Address>,
-        ) -> Result<BTreeMap<String, Address>, String> {
-            match utxo.rpc_client.clone() {
-                UtxoRpcClientEnum::Electrum(client) => {
-                    // Collect the scrpithash for every address into a map.
-                    let scripthash_to_address_map = addresses
-                        .into_iter()
-                        .map(|address| {
-                            let scripthash = address_to_scripthash(&address).map_err(|e| e.to_string())?;
-                            Ok((scripthash, address))
-                        })
-                        .collect::<Result<HashMap<String, Address>, String>>()?;
-                    // Add these subscriptions to the connection manager. It will choose whatever connections
-                    // it sees fit to subscribe each of these addresses to.
-                    client
-                        .connection_manager
-                        .add_subscriptions(&scripthash_to_address_map)
-                        .await;
-                    // Convert the hashmap back to btreemap.
-                    Ok(scripthash_to_address_map.into_iter().map(|(k, v)| (k, v)).collect())
-                },
-                UtxoRpcClientEnum::Native(_) => {
-                    Err("Balance streaming is currently not supported for native client.".to_owned())
-                },
-            }
-        }
-
-        if coin.as_ref().rpc_client.is_native() {
-            let msg = "Native RPC client is not supported for UtxoBalanceEventStreamer.";
-            ready_tx.send(Err(msg.to_string())).expect(RECEIVER_DROPPED_MSG);
-            panic!("{}", msg);
+        // Get all the addresses to subscribe to their balance updates.
+        let all_addresses = match coin.all_addresses().await {
```

Comment:
this is a file IO (DB querying) so it's quick enough. We can halt the response to `ready_tx` till this call resolves.

Reply:
thanks!
side Q: why is it an async call then?

Reply:
Accounts is a mutex, I need to recheck why this was needed again
https://github.com/KomodoPlatform/komodo-defi-framework/blob/9ea63ecb15f692b7c414999fdca0bf27b7845c99/mm2src/coins/hd_wallet/mod.rs#L292-L293

Reply:
Ok, because `HDAccountsMutex` is a mutex wrapping a map.

Reply:
aha i seen that. meant why is it async mutex to begin with. will check quickly if it could be made sync.

Reply:
was almost done, but came down to this call:
https://github.com/KomodoPlatform/komodo-defi-framework/blob/b68cb029b1ee12fafac8070d299da416ceb3bc47/mm2src/coins/hd_wallet/mod.rs#L432

we hold the accounts mutex while doing an async file IO call. but i guess that's needed for correctness. so the mutex indeed should by async 👍